components/movementsensor/wheeledodometry/wheeled_odometry.go

```diff
@@ -0,0 +1,341 @@
+// Package wheeledodometry implements an odometery estimate from an encoder wheeled base
+package wheeledodometry
+
+import (
+	"context"
+	"errors"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/edaniels/golog"
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	"go.viam.com/utils"
+
+	"go.viam.com/rdk/components/base"
+	"go.viam.com/rdk/components/motor"
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	rdkutils "go.viam.com/rdk/utils"
+)
+
+var model = resource.DefaultModelFamily.WithModel("wheeled-odometry")
+
+const (
+	defaultTimeInt = 500
+	oneTurn        = 2 * math.Pi
+)
+
+// Config is the config for a wheeledodometry MovementSensor.
+type Config struct {
+	LeftMotors        []string `json:"left_motors"`
+	RightMotors       []string `json:"right_motors"`
+	Base              string   `json:"base"`
+	TimeIntervalMSecs float64  `json:"time_interval_msecs,omitempty"`
+}
+
+type motorPair struct {
+	left  motor.Motor
+	right motor.Motor
+}
+
+type odometry struct {
+	resource.Named
+	resource.AlwaysRebuild
+
+	lastLeftPos        float64
+	lastRightPos       float64
+	baseWidth          float64
+	wheelCircumference float64
+	timeIntervalMSecs  float64
+
+	motors []motorPair
+
+	angularVelocity spatialmath.AngularVelocity
+	linearVelocity  r3.Vector
+	position        r3.Vector
+	orientation     spatialmath.EulerAngles
+
+	cancelFunc              func()
```

Comment:
[important] This never gets initialized: when you call it in `Close()`, I expect you to get a "can't call `nil` as a function" sort of error. On the other side, `trackPosition()` is given the main background context, which never gets canceled and is never done. To fix both issues, I suspect this is what `newWheeledOdometry()` should really do:
```go
backgroundContext, cancelFunc := context.WithCancel(context.Background())

o := &odometry{
    ...
    cancelFunc: cancelFunc,
}

o.trackPosition(backgroundContext)
```

Reply:
yeah I actually started the reconfigure implementation and noticed this as well. going to add the following lines to the end of the creator
```
ctx, cancelFunc := context.WithCancel(context.Background())
o.cancelFunc = cancelFunc
o.trackPosition(ctx)